app/src/tracing.h

```diff
@@ -0,0 +1,141 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#pragma once
+
+#include "constants.h"
+
+#include <time.h>
+#include <regex>
+#include <sstream>
+#include <functional>
+#include <ccf/endpoint_context.h>
+#include <ccf/base_endpoint_registry.h>
+
+namespace scitt
+{
+  std::string REQUEST_ID_HEADER = "x-ms-request-id";
+  std::string CLIENT_REQUEST_ID_HEADER = "x-ms-client-request-id";
+
+  std::regex CLIENT_REQUEST_ID_REGEX("^[0-9a-zA-Z-]+");
+
+  thread_local std::string request_id;
+  thread_local std::optional<std::string> client_request_id;
+
+  int diff_timespec_ms(const struct timespec& time0, const struct timespec& time1) {
+    return (time1.tv_sec - time0.tv_sec) * 1000
+        + (time1.tv_nsec - time0.tv_nsec) / 1000000;
+  }
+
+  std::string create_request_id() {
+    std::stringstream stream;
+    stream << std::hex << rand();
+    return stream.str();
+  }
+
+  inline void SCITT_INFO(const std::string& msg)
```

Comment:
We'll want these to be macros, which allows `__FILE__` and `__LINE__` to be accurate, and to get formatting syntax `SCITT_INFO("value={}", value)` working.

Unless [`source_location`](https://en.cppreference.com/w/cpp/utility/source_location) works on our compiler and we are willing to use CCF's logger directly to supply these, in which case we can get the formatting working using a variadic template.

Reply:
Done.